dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/jaxrspropagation/JaxEndpoints.java

```diff
@@ -0,0 +1,159 @@
+import jakarta.ws.rs.client.ClientBuilder;
+
+import org.eclipse.microprofile.rest.client.RestClientBuilder;
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+@ApplicationPath("")
+@Path("endpoints")
+public class JaxEndpoints extends Application {
+
+    @Inject
+    InMemorySpanExporter spanExporter;
+
+    @GET
+    @Path("/readspans")
+    public Response readSpans() {
+        List<SpanData> spanData = spanExporter.getFinishedSpanItems(3);
+
+        SpanData firstURL = spanData.get(0);
+        SpanData httpGet = spanData.get(1);
+        SpanData secondURL = spanData.get(2);
+
+        assertEquals(firstURL.getKind(), SERVER);
+        assertEquals(httpGet.getKind(), CLIENT);
+        assertEquals(secondURL.getKind(), SERVER);
+
+        assertEquals(httpGet.getParentSpanId(), firstURL.getSpanId());
+        assertEquals(secondURL.getParentSpanId(), httpGet.getSpanId());
+
+        return Response.ok("Test Passed").build();
+    }
+
+    @GET
+    @Path("/jaxclient")
+    public Response getJax(@Context UriInfo uriInfo) {
+        assertNotNull(Span.current());
+        try {
+            Thread.sleep(3000);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        spanExporter.reset();
+
+        Baggage.builder().put("foo", "bar").build().makeCurrent();
+        Baggage baggage = Baggage.current();
+        assertEquals("bar", baggage.getEntryValue("foo"));
+
+        Client client = ClientBuilder.newClient();
+        String url = new String(uriInfo.getAbsolutePath().toString());
+        url = url.replace("jaxclient", "jaxtwo"); //The jaxclient will use the URL as given so it needs the final part to be provided.
+
+        String result = client.target(url)
+            .request(MediaType.TEXT_PLAIN)
+            .get(String.class);
+
+        return Response.ok(result).build();
+    }
+
+    @GET
+    @Path("/jaxtwo")
+    public Response getJaxTwo() {
+        assertNotNull(Span.current());
+        Baggage result = Baggage.current();
+        assertEquals("bar", result.getEntryValue("foo"));
+        return Response.ok("Test Passed").build();
+    }
+
+    @GET
+    @Path("/mpclient")
+    public Response getMP(@Context UriInfo uriInfo) {
+        assertNotNull(Span.current());
+        try {
+            Thread.sleep(3000);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        spanExporter.reset();
+
+        Baggage.builder().put("foo", "bar").build().makeCurrent();
+        Baggage baggage = Baggage.current();
+        assertEquals("bar", baggage.getEntryValue("foo"));
+
+        String baseUrl = uriInfo.getAbsolutePath().toString().replace("/mpclient", ""); //The mpclient will add the final part of the URL for you, so we remove the final part.
+        URI baseUri = null;
+        try {
+            baseUri = new URI(baseUrl);
+        } catch (Exception e) {
+            fail("URI exception from " + baseUrl);
+        }
```

Comment:
Probably not a huge deal here, but calling `fail()` like this will throw away the original exception, which may have useful information in it.

Rethrowing as a `RuntimeException` and keeping the cause is probably better here, though if you want it to show up as a test _failure_ (rather than a test _exception_), throwing `AssertionError` will do that while also letting you keep the original exception as the cause.

Reply:
I also I think we should never be throwing away the cause when catching something general like `Exception`, even in a test. It's not quite as bad if you're catching something specific like `URISyntaxException`.